lib/state_machine/transition.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "message.cpp"
```

Comment:
Don't include a cpp file. Always include headers (all functions, classes, etc. should be defined there anyways).